compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java

```diff
@@ -915,4 +1006,67 @@ private boolean isMemberAllowed(SymbolEnv env, BSymbol symbol) {
                 && env.enclInvokable.symbol.receiverSymbol.type.tsymbol == symbol.owner
                 || isMemberAllowed(env.enclEnv, symbol));
     }
+
+    /**
+     * Returns the eligibility to use 'stamp' inbuilt function against the respective expression.
+     *
+     * @param targetType target type that 'stamp' function is used
+     * @return eligibility to use 'stamp' function
+     */
+    private boolean isStampSupportedForTargetType(BType targetType) {
+        if (types.isAnydata(targetType)) {
+            switch (targetType.tag) {
+                case TypeTags.ARRAY:
+                    //Primitive type array does not support stamp because primitive arrays aren't using ref registry.
+                    int arrayConstraintTypeTag = ((BArrayType) targetType).eType.tag;
+                    return !(arrayConstraintTypeTag == TypeTags.INT || arrayConstraintTypeTag == TypeTags.BOOLEAN ||
+                            arrayConstraintTypeTag == TypeTags.FLOAT || arrayConstraintTypeTag == TypeTags.BYTE ||
+                            arrayConstraintTypeTag == TypeTags.STRING || arrayConstraintTypeTag == TypeTags.DECIMAL);
+                case TypeTags.UNION:
+                case TypeTags.INT:
+                case TypeTags.BOOLEAN:
+                case TypeTags.STRING:
+                case TypeTags.FLOAT:
+                case TypeTags.DECIMAL:
+                case TypeTags.BYTE:
+                case TypeTags.TABLE:
+                    return false;
+                default:
+                    return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the eligibility whether stamp can be on the given value type.
+     *
+     * @param sourceType source type used for the stamp operation
+     * @return eligibility to use as the target type for 'stamp' function
+     */
+    private boolean isStampSupportedForSourceType(BType sourceType) {
+
+        if (types.isAnydata(sourceType)) {
+            switch (sourceType.tag) {
+                case TypeTags.ARRAY:
+                    // Primitive type array does not support stamp because primitive arrays are not using ref registry.
+                    int arrayConstraintTypeTag = ((BArrayType) sourceType).eType.tag;
+                    return !(arrayConstraintTypeTag == TypeTags.INT || arrayConstraintTypeTag == TypeTags.BOOLEAN ||
```

Comment:
We could check whether tag is less than some value to detect these all.

Reply:
Fixed.

Reply:
Will be incorporate with next iteration.